modules/util/tld-formatter/build.gradle

```diff
@@ -14,9 +14,9 @@ dependencies {
 	compile group: "log4j", name: "log4j", version: "1.2.17"
 	compile group: "org.dom4j", name: "dom4j", version: "2.1.3"
 
-	compileOnly group: "com.liferay", name: "com.liferay.petra.xml", version: "1.0.0"
+	compileOnly group: "com.liferay", name: "com.liferay.petra.string", version: "5.2.0"
 	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.impl", version: "2.0.0"
-	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "2.0.0"
+	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "2.49.0"
 	compileOnly group: "com.liferay.portal", name: "com.liferay.util.java", version: "2.5.0"
```

Comment:
@julschong Why you do not upgrade com.liferay.util.java here?
Also, why need to upgrade portal kernel version?

Reply:
@tinatian as we discussed,
kernel update is because the newly inline code in TLDFormatter caused error when I ran ant format-tld
```
format-tld: 
    [java] Exception in thread "main" java.lang.NoClassDefFoundError: com/liferay/portal/kernel/memory/SoftReferenceThreadLocal 
    [java]    at com.liferay.portal.kernel.util.StringUtil.replace(StringUtil.java:2655) 
    [java]    at com.liferay.portal.kernel.util.StringUtil.replace(StringUtil.java:2618) 
    [java]    at com.liferay.tld.formatter.TLDFormatter._documentToString(TLDFormatter.java:129) 
    [java]    at com.liferay.tld.formatter.TLDFormatter._formatTLD(TLDFormatter.java:162) 
    [java]    at com.liferay.tld.formatter.TLDFormatter.access$100(TLDFormatter.java:51) 
    [java]    at com.liferay.tld.formatter.TLDFormatter$1.visitFile(TLDFormatter.java:91) 
    [java]    at com.liferay.tld.formatter.TLDFormatter$1.visitFile(TLDFormatter.java:74) 
    [java]    at java.nio.file.Files.walkFileTree(Files.java:2670) 
    [java]    at java.nio.file.Files.walkFileTree(Files.java:2742) 
    [java]    at com.liferay.tld.formatter.TLDFormatter.<init>(TLDFormatter.java:72) 
    [java]    at com.liferay.tld.formatter.TLDFormatter.main(TLDFormatter.java:62) 
    [java] Caused by: java.lang.ClassNotFoundException: com.liferay.portal.kernel.memory.SoftReferenceThreadLocal 
    [java]    at java.net.URLClassLoader.findClass(URLClassLoader.java:382) 
    [java]    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) 
    [java]    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349) 
    [java]    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) 
    [java]    ... 11 more 
    [java] Java Result: 1
```

Reply:
But I do think that util.java needs to be updated as well, let me to some more testing locally to make sure